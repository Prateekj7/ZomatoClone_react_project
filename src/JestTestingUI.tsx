import { useState } from "react";

export default function JestTestingUI() {
    const [name, setName] = useState("");
    return (
        <>
           <input type="text" value={name} onChange={(e) => setName(e.target.value) } data-testid="input"/>
           <label data-testid="output">{`Hello ${name}`}</label>
        </>
    );
}
