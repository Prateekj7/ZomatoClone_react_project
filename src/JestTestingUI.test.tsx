import { fireEvent, render } from "@testing-library/react"
import JestTestingUI from "./JestTestingUI"

describe("JestTestingUI", () => {
    it("if label exist in the UI", () => {
        const {baseElement} = render(<JestTestingUI />);
        expect(baseElement).toBeInTheDocument();
    })
    it("Check if label is correct", () => {
        const {getByTestId} = render(<JestTestingUI />);
        const input = getByTestId("input");
        fireEvent.change(input, {target:{value:"geeks"}})
        const output = getByTestId('output');
        expect(output.innerHTML).toBe("Hello geeks");
    })
    it("Snapshot testing", () => {
        const {baseElement} = render(<JestTestingUI />);
        expect (baseElement).toMatchSnapshot();
    })
})