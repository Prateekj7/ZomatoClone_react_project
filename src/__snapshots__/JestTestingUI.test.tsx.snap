// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JestTestingUI Snapshot testing 1`] = `
<body>
  <div>
    <input
      data-testid="input"
      type="text"
      value=""
    />
    <label
      data-testid="output"
    >
      Hello 
    </label>
  </div>
</body>
`;
